src/main/kotlin/no/skatteetaten/aurora/boober/service/RedeployService.kt

```diff
@@ -24,55 +28,82 @@ class RedeployService(val openShiftClient: OpenShiftClient, val openShiftObjectG
         }
     }
 
-    fun triggerRedeploy(deploymentSpec: AuroraDeploymentSpec, redeployContext: RedeployContext): RedeployResult {
+    fun triggerRedeploy(deploymentSpec: AuroraDeploymentSpec, imageStream: ImageStream?, deploymentConfig: DeploymentConfig?): RedeployResult {
         val type = deploymentSpec.type
         if (type == TemplateType.build || type == TemplateType.development) {
```

Comment:
Hvis du ser kommentaren min om bruk av OpenShiftRequest i forbindelse med uthenting av ImageStream, og endrer dette til å kun bruke en vanlig get-request, vil du se at informasjonen man trenger fra deploymentspec'en begrenser seg til namespace og application name (dersom vi later som if-en denne kommentaren er festet til ikke eksisterer ;). Denne informasjonen er også tilgjengelig fra DeploymentConfigen (og også ImageStream?) direkte. Jeg mener mao. at det bør være mulig å lage en triggerRedeploy som kun tar DeploymentConfig og ImageStream som parametre.

TemplateType.build mener jeg ikke er i bruk, og kanskje @bjartek kan skyte inn om man evt. kan avgjøre om det er TemplateType development uten å se på AuroraDeploymentSpec'en? Evt. kan kanskje sjekken på om dette er development gjøres utenfor triggerRedeploy? Evt. at deploymentspecen kun brukes her, og ikke sendes med som parameter videre?

Reply:
Build er ikke i bruk det er helt riktig. Jeg synes fint vi kan sjekke for type før vi kaller metoden og da la være å sende inn hele ADC